docs/website/docs/general-usage/dataset-access/dataset.md

```diff
@@ -0,0 +1,238 @@
+
+#### As a Pandas DataFrame
+
+```py
+df = items_relation.df()
+```
+
+#### As a PyArrow Table
+
+```py
+arrow_table = items_relation.arrow()
+```
+
+#### As a list of Python tuples
+
+```py
+items_list = items_relation.fetchall()
+```
+
+## Lazy loading behavior
+
+The `ReadableDataset` and `ReadableRelation` objects are **lazy-loading**. This means that they do not immediately fetch data when you create them. Data is only retrieved when you perform an action that requires it, such as calling `.df()`, `.arrow()`, or iterating over the data. This approach optimizes performance and reduces unnecessary data loading.
+
+## Iterating over data in chunks
+
+To handle large datasets efficiently, you can process data in smaller chunks.
+
+### Iterate as Pandas DataFrames
+
+```py
+for df_chunk in items_relation.iter_df(chunk_size=500):
+    # Process each DataFrame chunk
+    pass
+```
+
+### Iterate as PyArrow Tables
+
+```py
+for arrow_chunk in items_relation.iter_arrow(chunk_size=500):
+    # Process each PyArrow chunk
+    pass
+```
+
+### Iterate as lists of tuples
+
+```py
+for items_chunk in items_relation.iter_fetch(chunk_size=500):
+    # Process each chunk of tuples
+    pass
+```
+
+The methods available on the ReadableRelation correspond to the methods available on the cursor returned by the SQL client. Please refer to the [SQL client](./sql-client.md#supported-methods-on-the-cursor) guide for more information.
+
+## Modifying queries
+
+You can refine your data retrieval by limiting the number of records, selecting specific columns, or chaining these operations.
+
+### Limit the number of records
+
+```py
+# Get the first 50 items as a PyArrow table
+arrow_table = items_relation.limit(50).arrow()
+```
+
+#### Using `head()` to get the first 5 records
+
+```py
+df = items_relation.head().df()
+```
+
+### Select specific columns
+
+```py
+# Select only 'col1' and 'col2' columns
+items_list = items_relation.select("col1", "col2").fetchall()
+
+# Alternate notation with brackets
+items_list = items_relation[["col1", "col2"]].fetchall()
+
+# Only get one column
+items_list = items_relation["col1"].fetchall()
+
+```
+
+### Chain operations
+
+You can combine `select`, `limit`, and other methods.
+
+```py
+# Select columns and limit the number of records
+arrow_table = items_relation.select("col1", "col2").limit(50).arrow()
+```
+
+## Supported destinations
+
+All SQL and filesystem destinations supported by `dlt` can utilize this data access interface. For filesystem destinations, `dlt` [uses **DuckDB** under the hood](./sql-client.md#the-filesystem-sql-client) to create views from Parquet or JSONL files dynamically. This allows you to query data stored in files using the same interface as you would with SQL databases. If you plan on accessing data in buckets or the filesystem a lot this way, it is advised to load data as parquet instead of jsonl, as **DuckDB** is able to only load the parts of the data actually needed for the query to work.
```

Comment:
```suggestion
All SQL and filesystem destinations supported by `dlt` can utilize this data access interface. For filesystem destinations, `dlt` [uses **DuckDB** under the hood](./sql-client.md#the-filesystem-sql-client) to create views from Parquet or JSONL files dynamically. This allows you to query data stored in files using the same interface as you would with SQL databases. If you plan on accessing data in buckets or the filesystem a lot this way, it is advised to load data as Parquet instead of JSONL, as **DuckDB** is able to only load the parts of the data actually needed for the query to work.
```